frappe/desk/doctype/form_tour/form_tour.js

```diff
@@ -2,36 +2,64 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on("Form Tour", {
-	setup: function (frm) {
-		if (!frm.doc.is_standard || frappe.boot.developer_mode) {
-			frm.trigger("setup_queries");
-		}
-	},
-
 	refresh(frm) {
 		if (frm.doc.is_standard && !frappe.boot.developer_mode) {
 			frm.trigger("disable_form");
 		}
-
-		frm.add_custom_button(__("Show Tour"), async () => {
-			const issingle = await check_if_single(frm.doc.reference_doctype);
-			let route_changed = null;
-
-			if (issingle) {
-				route_changed = frappe.set_route("Form", frm.doc.reference_doctype);
-			} else if (frm.doc.first_document) {
-				const name = await get_first_document(frm.doc.reference_doctype);
-				route_changed = frappe.set_route("Form", frm.doc.reference_doctype, name);
-			} else {
-				route_changed = frappe.set_route("Form", frm.doc.reference_doctype, "new");
+		frm.fields_dict["report_name"].get_query = function (doc) {
+			if (doc.reference_doctype) {
+				return {
+					filters: {
+						ref_doctype: doc.reference_doctype,
+					},
+				};
 			}
-			route_changed.then(() => {
-				const tour_name = frm.doc.name;
-				cur_frm.tour.init({ tour_name }).then(() => cur_frm.tour.start());
+			return {};
+		};
```

Comment:
You should move this to reference_doctype method also use frm.set_query()

Reply:
I have changed it to frm.set_query() but move this to reference_doctype method will break in some cases because reference_doctype is used in multiple places so if it is set before list_name is set to report_name filters won't apply.